app/scripts/views/sign_in.js

```diff
@@ -48,18 +48,28 @@ define(function (require, exports, module) {
     },
 
     beforeRender () {
-      this._account = this._suggestedAccount();
-
-      this._account.on('change:accessToken', () => {
-        // if no access token and password is not visible we need to show the password field.
-        if (! this._account.has('accessToken') && this.$('.password').is(':hidden')) {
-          // accessToken could be changed async by an external request after render
-          // If the ProfileClient fails to get an OAuth token with the current token then reset the view
-          this.chooserAskForPassword = true;
-          return this.render().then(function () {
-            this.setDefaultPlaceholderAvatar();
-          });
+      return p().then(() => {
+        const relier = this.relier;
+        // TODO - check how this interacts with form prefill.
+        const signinCode = relier.get('signinCode');
+        if (signinCode) {
+          return this.user.accountFromSigninCode(signinCode)
+            .then((account) => {
+              // The signinCode is consumed and can no
+              // longer be used. If the user goes to password reset
+              // and then comes back, the email will be stored in
+              // the FormPrefill model.
+              relier.unset('signinCode');
+              return account;
+            }, (err) => {
+              // log and swallow the error. Errors consuming a
+              // signinCode should not prevent the user from
+              // signing in normally.
+              this.logError(err);
```

Comment:
Add the new error code to lib/auth-errors.js @philbooth [added in the auth-server PR](https://github.com/mozilla/fxa-auth-server/pull/1906/files#diff-0f426376b964e609c5c4618eb8bed3b2R243).